server/tserver/src/main/java/org/apache/accumulo/tserver/replication/AccumuloReplicaSystem.java

```diff
@@ -288,15 +295,36 @@ public String execute(ReplicationCoordinator.Client client) throws Exception {
             }
           } else {
             span = Trace.start("WAL replication");
+
+            ExecutorService executor = Executors.newFixedThreadPool(1);
```

Comment:
Wasn't aware of that special Executor -- that'll work here, no need for another thread.  Thanks!